ethereumjs/ultralight
# Glados integration

This PR adds some tweaks to allow processing of post merge blocks (i.e. skipping proof validation of the header) and also adds a new bridge node script that pipes the latest blocks to an Ultralight node.

To test:
- Run the CLI client -- `npm run dev` from the `cli` package root
- Run the bridge node -- `npx ts-node --esm scripts/bridge.ts [my json rpc provider here]`.  You should start seeing output showing blocks being added to the Ultralight node's db

Running Glados to audit blocks:
- Clone the glados repo - https://github.com/ethereum/glados.git
- Create a sqlite database -- `touch /absolute/path/to/sql/database`
- Run `glados-monitor` -- 
```sh
RUST_LOG=glados_monitor=debug cargo run -p glados-monitor -- \
    --migrate \
    --database-url sqlite:////path/to/sqlite/database.sqlite \
    follow-head \
    --provider-url [myjsonrpcprovider.com] 
```
- Run `glados-audit`
```sh
RUST_LOG=glados_audit=debug cargo run -p glados-audit -- \    
    --database-url sqlite:////path/to/sqlite/database.sqlite \
--transport http \
--http-url http://localhost:8545
```
- Run `glados-web`
```sh
RUST_LOG=debug cargo run -p glados-web -- \
    --database-url sqlite:////path/to/sqlite/database.sqlite
```

Open a browser and go to `http://localhost:3001` to see the Glados UI
